src/main/java/com/example/bookstore/controler/BookController.java

```diff
@@ -18,40 +21,54 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book management",
+        description = "Manage books with CRUD operations and search functionality")
 @RestController
 @RequestMapping("/books")
 @RequiredArgsConstructor
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books",
+            description = "Get a list of all books with pagination support")
+    public List<BookDto> getAll(Pageable pageable) {
```

Comment:
you can also add these annotations
```suggestion
    public List<BookDto> getAll(@ParameterObject @PageableDefault Pageable pageable) {
```